sources/typescript/src/adapters/kaltura-parallel-upload-request-adapter.ts

```diff
@@ -0,0 +1,240 @@
+import { KalturaUploadRequest } from "../api/kaltura-upload-request";
+import { createEndpoint, prepareParameters } from "./utils";
+import { KalturaClientException } from "../api/kaltura-client-exception";
+import { KalturaRequestOptions } from "../api/kaltura-request-options";
+import { KalturaClientOptions } from "../kaltura-client-options";
+import { CancelableAction } from "../cancelable-action";
+import { KalturaUploadConnectionsManager } from "./kaltura-upload-connections-manager";
+import { KalturaUploadRequestAdapter } from "./kaltura-upload-request-adapter";
+
+interface UploadByParallelChunksData {
+    chunkUploadEnabled: boolean;
+
+    /**
+     * bytes uploaded so far
+     */
+    uploaded: number;
+
+    /**
+     * total number of chunks in the current file
+     */
+    totalChunks: number;
+
+    /**
+     * number of bytes per chunk
+     */
+    chunkSize: number;
+
+    /**
+     * number of chunks done uploading
+     */
+    chunksUploaded: number;
+
+    /**
+     * number of the next chunk to start uploading
+     */
+    nextChunkIndex: number;
+}
+
+export class KalturaParallelUploadRequestAdapter extends KalturaUploadRequestAdapter {
+
+    constructor(clientOptions: KalturaClientOptions, defaultRequestOptions: KalturaRequestOptions) {
+        super(clientOptions, defaultRequestOptions);
+        if (clientOptions.parallelUploadsDisabled) {
+            return;
+        }
+        if (KalturaUploadConnectionsManager.getTotalConnections() === -1) {
```

Comment:
i take it -1 is the initial value indicating no value yet?
worth mentioning it a comment above.